### ISSUE TITLE
Accessing TYPES through instance variables

### ISSUE THREAD

**Issue by larshp** [SAP/styleguides#288]
For a section in the guide suggest adding "Don't access TYPES through instance variables"

Background: same story as https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#dont-call-static-methods-through-instance-variables

anti pattern example,

```abap
CLASS lcl DEFINITION.
  PUBLIC SECTION.
    TYPES foo TYPE i.
ENDCLASS.
CLASS lcl IMPLEMENTATION.
ENDCLASS.

INTERFACE lif.
  DATA ref TYPE REF TO lcl.
  TYPES blah TYPE ref->foo.   " <- this is bad
ENDINTERFACE.
```

instead do,

```abap
CLASS lcl DEFINITION.
  PUBLIC SECTION.
    TYPES foo TYPE i.
ENDCLASS.
CLASS lcl IMPLEMENTATION.
ENDCLASS.

INTERFACE lif.
  DATA ref TYPE REF TO lcl.
  TYPES blah TYPE lcl=>foo.   " <- this is good
ENDINTERFACE.
```

any thoughts/comments?

**Comment by nununo** [comment]
Agree!

**Comment by N2oB6n-SAP** [comment]
I agree, even though (this holds true for the referenced section on instance variables as well)

> [...] is a potential source of confusion [...]

feels like a rather soft and/or incomplete argument to me.

I would prefer to actually name the confusion (the accessed type not being instance-specific/dependent) and possibly argue that refactoring of the instance variable [name] would misleadingly touch a piece of code that is unrelated to the type definition.

@larshp Would you like to open a PR yourself?
